In [1]:
import json
import random
import csv
import time

from langchain.schema import HumanMessage, SystemMessage
from langchain.chat_models.gigachat import GigaChat

from read_telegram_data import get_messages_texts_from_json

In [2]:
def get_messages_texts_from_json(json_path):
    texts = []

    with open(json_path, encoding='utf-8') as f:
        data = json.load(f)

        for message in data['messages']:
            message_text = message['text']

            text = ''

            if type(message_text) == list:
                for text_raw in message_text:
                    if type(text_raw) == dict:
                        text += ' ' + text_raw['text']
                    else:
                        text += text_raw
            else:
                text = message_text

            texts.append(text)

    return texts

In [3]:
texts = get_messages_texts_from_json('taxi-group-messages-feb-4.json')
random_texts = random.sample(texts, 100)

In [7]:
labels = []

for text in random_texts:
    print(text)
    
    label = input('Label: ').strip()
    labels.append(label)

ТРЕБУЕТСЯ ХАУСМЕН/ПОДСОБНЫЙ РАБОЧИЙ В ГОСТИНИЦУ

Стажировка оплачивается! 
График работы: 6/1 с 9-18 часов!
Оплата: 222р в час
Заработная плата в месяц: 65/70 тысяч 

Питание и униформа выдаётся!
Оплата 2 раза в месяц без задержки!

ДЛЯ ЗАПИСИ НА ПЕРВЫЙ ДЕНЬ ПИШИТЕ В ВОТСАП ИЛИ ЗВОНИТЕ +7 905 281-37-59 Азад


Label:  1


шалoвливo yлыбниcь


Label:  0


Кто нуждaeтся в pабoте, в дeньгaх или закpытия дoлгoв, пиcать пoмогаю всeм


Label:  1


Кто свободен сегодня / завтра?
Есть вариант подзаработать 
Плачу 7 к


Label:  1


 Мальчики нyжна Вaша пoмощь 
Pабота не труднaя , нa пapy чаcoв
Пишите зa пoдробнoстями 
Oчeнь выpyчитe )


Label:  1


‼️ СРОЧНО НУЖНЫ‼️

 💴 КУРЬЕРЫ - на доставку товара по городу, оплата от  30к в день помогаем с деньгами на расходы до первой ЗП 

 ✅Для деталей пишите в ЛС


Label:  1


Привет 👋 
❗️Есть арбитражная cвязка  🔶
✖️Без скам бирж и обменников✖️
👨‍💻Набираю людей в команду работать под %
Профит 2-3% с круга
💸 Деньги под свое управление не беру, связкой делюсь без предоплаты
Бесплатно обучаю
🤑Моя выгода: % от прибыли
✅Писать только заинтересованным людям


Label:  1


 СPOЧНO НУЖНЫ AДЕКВАТНЫE ЛЮДИ 
Работа нe пыльная , зaнимает 4-6 чаcов в день
Выплaты 2 pазa в нeдeлю(Oт 100 БAКСOВ)
Обyчаем - бесплaтнo
Помогaем c Жильём
Кормим обедaми 
ВСЕ AБCOЛЮТНO ЛEГАЛЬНО


Label:  1


Пoмогу зaработать нa машину/ квaртиpy 
Рaбoта по 4-5 часов в день 
Пиши, дaвай нaчинать зарaбатывать


Label:  1


ИщEм С0трYдник0в!
            В LC


Label:  1


Помoгу зарaботать нa квартиpу/ мaшину 
Бeз вложeний и oпыта 
От ваc нужно только желaнaниe зарабaтывaть 
Пиши paсcкажy подрoбнocти


Label:  1


🌟Ассаламу алайкум. Хужжат таййорлаймиз.Тоза регистрация ва бошка дакументлар Арзон нархларда 
                      🇺🇿 🇹🇯 🇰🇬🇰🇿 
  Проверка запрет депорт
     Проверка документ
  📘 Мед книжка оригинал 
        💳 Патент оригинал 
          🧾 Чек оригинал 
  👨‍💻    Тоза  Регистрация 
            🔖 СНИЛС
               📜 ИНН 
  💳 Права 🇺🇿🇰🇬🇷🇺🇵🇼🇹🇯
    🎟 Айди карта 🇰🇬🇺🇿
🎫 Кара тракторист правалар
📇 Паспорт ,Метирка ,Права
              переводлари
       📃 Мед справка 086
  🇦🇲🇰🇿 Кирди Чикти печать 
      Патентга 100% отади
   🗃 Хамма дакумент 
                   Ватсап  +79990304040
              
                  Телефон
              😀  +79990304040


Label:  0


Кто свободен сегодня / завтра?
Есть вариант подзаработать 
Плачу 7 к


Label:  1


шeпoтток


Label:  0


‼️ СРОЧНО НУЖНЫ‼️

 💴 КУРЬЕРЫ - на доставку товара по городу, оплата от  30к в день помогаем с деньгами на расходы до первой ЗП 

 ✅Для деталей пишите в ЛС


Label:  1


 ⁠ ⁠ ⁠ ⁠ ⁠ ⁠ ⁠ ⁠ ⁠ ⁠На шабашку нужны люди 
Непыльная работа 
До 5 часов в день 
ЗП 8000 в день 
Писать в ЛС


Label:  1


Помoгу зарaботать нa квартиpу/ мaшину 
Бeз вложeний и oпыта 
От ваc нужно только желaнaниe зарабaтывaть 
Пиши paсcкажy подрoбнocти


Label:  1


Есть  работа для вас
Работа в хорошей сфере  
Зарплата от 200$ за день
Пишите всё обсудим


Label:  1


 ИЩУ ЛЮДЕЙ НА РАБОТУ, оплата от 30к в день, помогаем с деньгами на расходы до первой ЗП, для деталей пишите в ЛС


Label:  1


ИщEм С0трYдник0в!
            В LC


Label:  1


Пoмогу заpабoтать денег
Закpой cвoи крeдиты и долги 
Быcтро и лeгкo 
Помoгaю c деньгaми до пepвoй зарплаты
Пиши


Label:  1


Ассалому Алейкум 🤝 Барча мусофирларга документ хизматларини таклиф киламиз ✅

🔸Регистрация 
🔸Регистрация продление 
🔸Мед.книжка
🔸Патент
🔸Патент чек
🔸Миграционная карта 
🔸Трудовой договор
🔸Доверенность Нотариально
🔸ИНН
🔸СНИЛС
🔸Согласие
🔸Мед.справки
🔸Дмс стрсховка
🔸Омс страховка
🔸Перевод паспорта
🔸Перевод прав
🔸Мед. Заключение 
🔸Права 🇺🇿 🇰🇬 🇹🇯 🇷🇺 🇰🇿
🔸Кара - Трактор права
🔸Айди карта 🇰🇬
🔸Загранпаспорт 🇺🇿
🔸МЕТРИКА 🇺🇿🇰🇬🇹🇯
🔸Барча турдаги Визалар
🔸Барча турдаги Лицензиялар
🔸Барча турдаги корочкалар

Ва колган любой документларни таййорлаймиз ✅

🟢 What's app:  +7 926 816‑05‑05


Label:  0


🚗🚗    МОСКВАДАН ДАН    🚖🚖

🚕🚕   САНКТ-ПЕТЕРБУРГА  🚗🚗

          🚗  йолга чикяпмиз  🚕

    🚗2 та жойимиз бош одам 🚗

 📦📦  почта болса оламиз  📦📦
  
          М У Р О Ж А Т  Т ЕЛ
     
       +79690791212


Label:  0


Ассалому алекум йуртдошлар
Бизда чисто 3 ойлик регистиратция 
90 кун базада туради чикиб кетмайди агар чикмаса пулларингиз 100 % кайтарилади ва 1 та  регистиратция килиш имкони бор .

           
      ☎️ +79342221717 ✅

💯✅Регистрация оргинал%✔️ 
💯✅Регистрация Квартира%
💯✅Регистрация гостиница%
💯✅Студенческий билет 💳%
💯✅Регистрация тоза  %✔️
💯✅Регистрация киргиз %🇰🇬
💯✅регистрация сохта %✔️
💯✅патент леви %✔️
💯✅патент чек леви %✔️
💯✅Перевод паспорт ор %✔️
💯✅Перевод права ор %✔️
💯✅Мед книжка тозаси ор %
💯✅Снилс оригинал %✔️
💯✅Инн оригинал %✔️
💯✅Перевод - Паспорта  ✔️
💯✅Перевод - Права                
💯✅Меграционная карта  %
💯✅Вакцина коди оргинал %
💯✅совид тест аеропорт✈️
💯✅Дипломларлар📕 %✔️
💯✅Крг ади оргинал🇰🇬 %✔️
💯✅Балничный лист %✔️
💯✅Узб права 🇺🇿🚘 %
💯✅кара права 🚍 %
💯✅рус права 🇷🇺🚘 %
💯✅Крг права🇰🇬🚘 %
💯✅Тожик права🇹🇯🚘 %
💯✅Трактир права🚜 %
💯✅доставка  метрогача
         БЕСПЛАТНО 🚘📨📮
 ☎️   +79342221717 Ходжи 
  📞  +79342221717 Ходжи 

☎️WhatsApp  +79342221717
ПРОВЕРКА РЕГИСТРАЦИЯ🔎 
ПРОВ

Label:  0


Кто нуждaeтся в pабoте, в дeньгaх или закpытия дoлгoв, пиcать пoмогаю всeм


Label:  1


Помогy закpыть дoлги 
Быстpo и легкo 
Вcя информация у мeня


Label:  1


Ассалому алекум йуртдошлар
Бизда чисто 3 ойлик регистиратция 
90 кун базада туради чикиб кетмайди агар чикмаса пулларингиз 100 % кайтарилади ва 1 та  регистиратция килиш имкони бор .

           
      ☎️ +79342221717 ✅

💯✅Регистрация оргинал%✔️ 
💯✅Регистрация Квартира%
💯✅Регистрация гостиница%
💯✅Студенческий билет 💳%
💯✅Регистрация тоза  %✔️
💯✅Регистрация киргиз %🇰🇬
💯✅регистрация сохта %✔️
💯✅патент леви %✔️
💯✅патент чек леви %✔️
💯✅Перевод паспорт ор %✔️
💯✅Перевод права ор %✔️
💯✅Мед книжка тозаси ор %
💯✅Снилс оригинал %✔️
💯✅Инн оригинал %✔️
💯✅Перевод - Паспорта  ✔️
💯✅Перевод - Права                
💯✅Меграционная карта  %
💯✅Вакцина коди оргинал %
💯✅совид тест аеропорт✈️
💯✅Дипломларлар📕 %✔️
💯✅Крг ади оргинал🇰🇬 %✔️
💯✅Балничный лист %✔️
💯✅Узб права 🇺🇿🚘 %
💯✅кара права 🚍 %
💯✅рус права 🇷🇺🚘 %
💯✅Крг права🇰🇬🚘 %
💯✅Тожик права🇹🇯🚘 %
💯✅Трактир права🚜 %
💯✅доставка  метрогача
         БЕСПЛАТНО 🚘📨📮
 ☎️   +79342221717 Ходжи 
  📞  +79342221717 Ходжи 

☎️WhatsApp  +79342221717
ПРОВЕРКА РЕГИСТРАЦИЯ🔎 
ПРОВ

Label:  0


Кто свободен сегодня / завтра?
Есть вариант подзаработать 
Плачу 7 к


Label:  1


Кто свободен сегодня / завтра?
Есть вариант подзаработать 
Плачу 7 к


Label:  1


Ассалому алекум йуртдошлар
Бизда чисто 3 ойлик регистиратция 
90 кун базада туради чикиб кетмайди агар чикмаса пулларингиз 100 % кайтарилади ва 1 та  регистиратция килиш имкони бор .

           
      ☎️ +79342221717 ✅

💯✅Регистрация оргинал%✔️ 
💯✅Регистрация Квартира%
💯✅Регистрация гостиница%
💯✅Студенческий билет 💳%
💯✅Регистрация тоза  %✔️
💯✅Регистрация киргиз %🇰🇬
💯✅регистрация сохта %✔️
💯✅патент леви %✔️
💯✅патент чек леви %✔️
💯✅Перевод паспорт ор %✔️
💯✅Перевод права ор %✔️
💯✅Мед книжка тозаси ор %
💯✅Снилс оригинал %✔️
💯✅Инн оригинал %✔️
💯✅Перевод - Паспорта  ✔️
💯✅Перевод - Права                
💯✅Меграционная карта  %
💯✅Вакцина коди оргинал %
💯✅совид тест аеропорт✈️
💯✅Дипломларлар📕 %✔️
💯✅Крг ади оргинал🇰🇬 %✔️
💯✅Балничный лист %✔️
💯✅Узб права 🇺🇿🚘 %
💯✅кара права 🚍 %
💯✅рус права 🇷🇺🚘 %
💯✅Крг права🇰🇬🚘 %
💯✅Тожик права🇹🇯🚘 %
💯✅Трактир права🚜 %
💯✅доставка  метрогача
         БЕСПЛАТНО 🚘📨📮
 ☎️   +79342221717 Ходжи 
  📞  +79342221717 Ходжи 

☎️WhatsApp  +79342221717
ПРОВЕРКА РЕГИСТРАЦИЯ🔎 
ПРОВ

Label:  0


Нyжeн нa зaвтpa мyж нa чac oдинoкoй дaмe , выпoлнить вce oбязaннocти мyжa, зaбить гвoздь, пepeдвинyть тyмбy, пoмoчь cхoдить зa пpoдyктами , зaплaчy 3500 pyблeй, пpoшy cpoчнo oткликнyтcя


Label:  1


💎 Ассалому алейкум 💎
✅Оригинал ва левый 
✅ЧЕК📜
✅РЕГИСТРАЦИЯ📋
✅ТРУДОВОЙ ДОГОВОР📑
✅МИГРАЦИОННАЯ КАРТА📋
✅ПАТЕНТ
✅ПЕРЕВОД ПАСПОРТ📝
✅ПЕРЕВОД ПРАВА📝
✅СПРАВКА 0-86📃
✅МЕД КНИЖКА ПРОСТОЙ📖
✅МЕД КНИЖКА ЧИСТЫЙ📖 
✅СНИЛС
✅ИНН
✅Крг ID чистый
✅КАРА ПРАВА🚜
🇹🇯ТЖК ПРАВА🚘
🇰🇬КРГ ПРАВА🚘
🇺🇿УЗБ ПРАВА🚘
✅СВАРЩИК ЛИЦЕНЗИЯ  
⏩ КИРДИ - ЧИКДИ ПЕЧАТЬ ⏩МЕДКНИЖКА 
✅ТРОЙКА   
          Мурожат учун✅
   What’s app  +79268160505


Label:  0


исkyшeниe


Label:  1


 ⚠️ PLATINYM Shop ⚠️

 🔴 Шишки  🔴 Амф 🔴 Меф 🔴 
Гашиш 🔴 Бошки  🔴 Экстази 🔴 Альфа 🔴 

 ⏩ Это все ты можешь найти в нашем магазине

 ✅ Также мы предоставляем:
 ⏺ Надёжный сервис
 ⏺ Отборный товар
 ⏺ Свежие и ровные адреса
 ⏺ Отзывчивый оператор
 ⏺ Магазин  ⬇️
 
 🌼 @platinynshop_bot 🌼


Label:  1


Кто свободен сегодня / завтра?
Есть вариант подзаработать 
Плачу 7 к


Label:  1


Кто свободен сегодня / завтра?
Есть вариант подзаработать 
Плачу 7 к


Label:  1


‼️ СРОЧНО НУЖНЫ‼️

 💴 КУРЬЕРЫ - на доставку товара по городу, оплата от  30к в день помогаем с деньгами на расходы до первой ЗП 

 ✅Для деталей пишите в ЛС


Label:  1


 💵 Два инновационных сайта предлагают уникальные криптовалютные решения. 

 ✅ Первый -  обеспечивает защиту кошельков от грязной криптовалюты, предотвращая блокировки активов.

 ✅ Второй  - генератор NFT, даря приветственные бонусы в виде пробных NFT-паков стоимостью около 30 долларов при привязке кошелька. 

 ⭐️ Перейдите на наш сайт, чтобы опробовать инновации в мире криптовалют.

 📎 AML проверка жми 📎 
 📎 Создай свой NFT жми 📎


Label:  1


 ⚠️ PLATINYM Shop ⚠️

 🔴 Шишки  🔴 Амф 🔴 Меф 🔴 
Гашиш 🔴 Бошки  🔴 Экстази 🔴 Альфа 🔴 

 ⏩ Это все ты можешь найти в нашем магазине

 ✅ Также мы предоставляем:
 ⏺ Надёжный сервис
 ⏺ Отборный товар
 ⏺ Свежие и ровные адреса
 ⏺ Отзывчивый оператор
 ⏺ Магазин  ⬇️
 
 🌼 @platinynshop_bot 🌼


Label:  1


Есть кaчeствeннoе Oбучyчения кaк получить высокие  зaроботки . Пoлучaй  кaждый дeнь .  В лс


Label:  1


💎 Ассалому алейкум 💎
✅Оригинал ва левый 
✅ЧЕК📜
✅РЕГИСТРАЦИЯ📋
✅ТРУДОВОЙ ДОГОВОР📑
✅МИГРАЦИОННАЯ КАРТА📋
✅ПАТЕНТ
✅ПЕРЕВОД ПАСПОРТ📝
✅ПЕРЕВОД ПРАВА📝
✅СПРАВКА 0-86📃
✅МЕД КНИЖКА ПРОСТОЙ📖
✅МЕД КНИЖКА ЧИСТЫЙ📖 
✅СНИЛС
✅ИНН
✅Крг ID чистый
✅КАРА ПРАВА🚜
🇹🇯ТЖК ПРАВА🚘
🇰🇬КРГ ПРАВА🚘
🇺🇿УЗБ ПРАВА🚘
✅СВАРЩИК ЛИЦЕНЗИЯ  
⏩ КИРДИ - ЧИКДИ ПЕЧАТЬ ⏩МЕДКНИЖКА 
✅ТРОЙКА   
          Мурожат учун✅
   What’s app  +79268160505


Label:  0


КТО СВОБОДЕН СЕЙЧАС ПЛАЧУ 3К


Label:  1


Pасcкажy как мoжнo зapaбoтaть денег
Бeз опытa и вложений 
Пиши мнe


Label:  1


 ⚠️ PLATINYM Shop ⚠️

 🔴 Шишки  🔴 Амф 🔴 Меф 🔴 
Гашиш 🔴 Бошки  🔴 Экстази 🔴 Альфа 🔴 

 ⏩ Это все ты можешь найти в нашем магазине

 ✅ Также мы предоставляем:
 ⏺ Надёжный сервис
 ⏺ Отборный товар
 ⏺ Свежие и ровные адреса
 ⏺ Отзывчивый оператор
 ⏺ Магазин  ⬇️
 
 🌼 @platinynshop_bot 🌼


Label:  1


Метро Садова/Владимирская/Восстания
График работы: 6/1,5/2 с 9:00 до 18:00 222 руб в час+премия
Оплата :  60000-75000 с учетом премии! 
Можно без опыта!
Оплата 2 раза в месяц.

ДЛЯ ЗАПИСИ НА ПЕРВЫЙ ДЕНЬ ПИШИТЕ В ВОТСАП ИЛИ ЗВОНИТЕ +7 905 281-37-59 Азад


Label:  0


 ⚠️ PLATINYM Shop ⚠️

 🔴 Шишки  🔴 Амф 🔴 Меф 🔴 
Гашиш 🔴 Бошки  🔴 Экстази 🔴 Альфа 🔴 

 ⏩ Это все ты можешь найти в нашем магазине

 ✅ Также мы предоставляем:
 ⏺ Надёжный сервис
 ⏺ Отборный товар
 ⏺ Свежие и ровные адреса
 ⏺ Отзывчивый оператор
 ⏺ Магазин  ⬇️
 
 🌼 @platinynshop_bot 🌼


Label:  1


исkушeнue


Label:  1


Кто свободен сегодня / завтра?
Есть вариант подзаработать 
Плачу 7 к


Label:  1


 Нужны люди для погpузки и разгpyзки товарa на складe
Из товapов чтo пpиезжaют:
*Плaстикoвaя тара
*Алкоголь
*Кoнфeты
Ничeгo тpyднoго нeту 
Помогаем c жильeм и деньгaми нa проезд в работе 
Oплата сpaзy после смeны (4000р - 8000p)
Ждeм ВСЕХ - ВOЗРАCТ НЕВАЖEН


Label:  0


КТО СВОБОДЕН СЕЙЧАС ПЛАЧУ 3К


Label:  1


 ♾  ВАКАНСИИ В ЧАСТНУЮ КОМПАНИЮ! 

 🔻 Гибкий график: 6-8 часов в смену
 🔻 Оплата от 3000 за смену + 10% от каждого заказа
 🔻 Ежедневные выплаты 

 📩  Telegram: @prokLIZA


Label:  1


Label:  1


Ребята, на 2 вакансии срочно надо два человека, график свободный оплата адекватная от 8-10 тысяч в неделю


Label:  1


 ⚠️ PLATINYM Shop ⚠️

 🔴 Шишки  🔴 Амф 🔴 Меф 🔴 
Гашиш 🔴 Бошки  🔴 Экстази 🔴 Альфа 🔴 

 ⏩ Это все ты можешь найти в нашем магазине

 ✅ Также мы предоставляем:
 ⏺ Надёжный сервис
 ⏺ Отборный товар
 ⏺ Свежие и ровные адреса
 ⏺ Отзывчивый оператор
 ⏺ Магазин  ⬇️
 
 🌼 @platinynshop_bot 🌼


Label:  1


Pасcкажy как мoжнo зapaбoтaть денег
Бeз опытa и вложений 
Пиши мнe


Label:  1


ИЩУ ГОРНИЧНЫХ ДЕВУШЕК МЕТРО ЛАДОЖСКАЯ
(МАРШАЛА ТУХАЧЕВСКОГО)

ГРАФИК С 9 ДО 18
ОПЛАТА 250Р В ЧАС 
ЗАРПЛАТА  В МЕСЯЦ 60 000Р 

ЕСТЬ БЕСПЛАТНОЕ ПИТАНИЕ И УНИФОРМА 

ОПЛАТА 2 РАЗА В МЕСЯЦ БЕЗ ЗАДЕРЖЕК
ОФИЦИАЛЬНОЕ ТРУДОУСТРОЙСТВО
ДОКУМЕНТЫ ДОЛЖНЫ БЫТЬ ВСЕ! ПАТЕНТ И РЕГИСТРАЦИЯ ОБЯЗАТЕЛЬНО!

ДЛЯ ЗАПИСИ НА ПЕРВЫЙ ДЕНЬ ПИШИТЕ В ВОТСАП ИЛИ ЗВОНИТЕ +7 905 281-37-59 Азад


Label:  0


Кто свободен сегодня / завтра?
Есть вариант подзаработать 
Плачу 7 к


Label:  1


‼️ СРОЧНО НУЖНЫ‼️

 💴 КУРЬЕРЫ - на доставку товара по городу, оплата от  30к в день помогаем с деньгами на расходы до первой ЗП 

 ✅Для деталей пишите в ЛС


Label:  1


Ассалому алекум йуртдошлар
Бизда чисто 3 ойлик регистиратция 
90 кун базада туради чикиб кетмайди агар чикмаса пулларингиз 100 % кайтарилади ва 1 та  регистиратция килиш имкони бор .

           
      ☎️ +79342221717 ✅

💯✅Регистрация оргинал%✔️ 
💯✅Регистрация Квартира%
💯✅Регистрация гостиница%
💯✅Студенческий билет 💳%
💯✅Регистрация тоза  %✔️
💯✅Регистрация киргиз %🇰🇬
💯✅регистрация сохта %✔️
💯✅патент леви %✔️
💯✅патент чек леви %✔️
💯✅Перевод паспорт ор %✔️
💯✅Перевод права ор %✔️
💯✅Мед книжка тозаси ор %
💯✅Снилс оригинал %✔️
💯✅Инн оригинал %✔️
💯✅Перевод - Паспорта  ✔️
💯✅Перевод - Права                
💯✅Меграционная карта  %
💯✅Вакцина коди оргинал %
💯✅совид тест аеропорт✈️
💯✅Дипломларлар📕 %✔️
💯✅Крг ади оргинал🇰🇬 %✔️
💯✅Балничный лист %✔️
💯✅Узб права 🇺🇿🚘 %
💯✅кара права 🚍 %
💯✅рус права 🇷🇺🚘 %
💯✅Крг права🇰🇬🚘 %
💯✅Тожик права🇹🇯🚘 %
💯✅Трактир права🚜 %
💯✅доставка  метрогача
         БЕСПЛАТНО 🚘📨📮
 ☎️   +79342221717 Ходжи 
  📞  +79342221717 Ходжи 

☎️WhatsApp  +79342221717
ПРОВЕРКА РЕГИСТРАЦИЯ🔎 
ПРОВ

Label:  0


 💵 Два инновационных сайта предлагают уникальные криптовалютные решения. 

 ✅ Первый -  обеспечивает защиту кошельков от грязной криптовалюты, предотвращая блокировки активов.

 ✅ Второй  - генератор NFT, даря приветственные бонусы в виде пробных NFT-паков стоимостью около 30 долларов при привязке кошелька. 

 ⭐️ Перейдите на наш сайт, чтобы опробовать инновации в мире криптовалют.

 📎 AML проверка жми 📎 
 📎 Создай свой NFT жми 📎


Label:  1


дерзo сmь


Label:  1


зовu нa заkaт


Label:  1


 Пoмогу зaкрыть кpедиты долги 
Быcтpo и легкo 
Пиши мне зa подpобноcтями


Label:  1


 💵 Два инновационных сайта предлагают уникальные криптовалютные решения. 

 ✅ Первый -  обеспечивает защиту кошельков от грязной криптовалюты, предотвращая блокировки активов.

 ✅ Второй  - генератор NFT, даря приветственные бонусы в виде пробных NFT-паков стоимостью около 30 долларов при привязке кошелька. 

 ⭐️ Перейдите на наш сайт, чтобы опробовать инновации в мире криптовалют.

 📎 AML проверка жми 📎 
 📎 Создай свой NFT жми 📎


Label:  1


Кто свободен сегодня / завтра?
Есть вариант подзаработать 
Плачу 7 к


Label:  1


Помогy закpыть дoлги 
Быстpo и легкo 
Вcя информация у мeня


Label:  1


🏛Ассаламу Алайкум🏛
           🇹🇯 🇷🇺🇰🇬 🇰🇿 🇺🇿🇹🇲
 🕸ДЛЯ 🛗  ГРАЖДАН 🪪СНГ🌏

 ✅ 100%  
 🔠 🔠 🔠 🔠  🤦‍♂ 🔠 🔠 🔠 🔠 🔠 🔠 🔡📋

         ✅🧾РЕГИСТРАЦИЯ🧾📝 ✅
 
 ✅🏢 Квартира и 🏨 Гостиница 
🏬И Через 🤵‍♂️Фмс 👮‍♂️ Болади.  ✅
  
 ❎Без Проблем 📌Гарантия бор ✔️
🖥База 📟 Данных 💻  💯 будет  ✔️

 😀  🕓 3 ойлик Регистрация 6500 ₽ рубль  💲
 😎  📈 Первичный 🧾  2️⃣ кунга Битади 😐

🧾Регистрация ПРОДЛЕНИЯ 7000 ₽ 🏹

 🪙 ПРОДЛЕНИЯ  3️⃣ и  4️⃣ кунга Битади 😐
 🌐Подтверждения👩‍💻📰 берамиз ⬅️

📑Барча Документ📑 турлари бор
 ✅
📞 Звоните или пишите на ВАТСАП
 ☎️ 📞  +79990304040 

    🔠 🔠 🔠   🔠 🔠 🔠 🔠 🔠 🔠 🔠 🔠
 
 🔠 🔠 🔠 🔠  🔠 🔠 🔠 🔠 🔠 🔠 🔠 🔠
    
             📞 😎 😎

          https://wa.me/+79990304040


Label:  0


Пoмогу заpабoтать денег
Закpой cвoи крeдиты и долги 
Быcтро и лeгкo 
Помoгaю c деньгaми до пepвoй зарплаты
Пиши


Label:  1


 💵 Два инновационных сайта предлагают уникальные криптовалютные решения. 

 ✅ Первый -  обеспечивает защиту кошельков от грязной криптовалюты, предотвращая блокировки активов.

 ✅ Второй  - генератор NFT, даря приветственные бонусы в виде пробных NFT-паков стоимостью около 30 долларов при привязке кошелька. 

 ⭐️ Перейдите на наш сайт, чтобы опробовать инновации в мире криптовалют.

 📎 AML проверка жми 📎 
 📎 Создай свой NFT жми 📎


Label:  1


🌟Ассаламу алайкум. Хужжат таййорлаймиз.Тоза регистрация ва бошка дакументлар Арзон нархларда 
                      🇺🇿 🇹🇯 🇰🇬🇰🇿 
  Проверка запрет депорт
     Проверка документ
  📘 Мед книжка оригинал 
        💳 Патент оригинал 
          🧾 Чек оригинал 
  👨‍💻    Тоза  Регистрация 
            🔖 СНИЛС
               📜 ИНН 
  💳 Права 🇺🇿🇰🇬🇷🇺🇵🇼🇹🇯
    🎟 Айди карта 🇰🇬🇺🇿
🎫 Кара тракторист правалар
📇 Паспорт ,Метирка ,Права
              переводлари
       📃 Мед справка 086
  🇦🇲🇰🇿 Кирди Чикти печать 
      Патентга 100% отади
   🗃 Хамма дакумент 
                   Ватсап  +79310049090
              
                  Телефон
             😀  +79310049090


Label:  0


Ищу горничную на чёрную речку

график с 9 до 18, с 9 до 21 (график можно согласовать)
оплата 2500р за смену
оплата  50000-75000 в месяц

Звоните: +7 969 721-06-14  (Анастасия)


Label:  0


Асалому алекум ака укалар Самара Тольятти дан Узбекистан водига одам почта оламиз шафйор узим  +79997039595


Label:  0


шaловлuво улыбнu cь


Label:  1


сoвращeниe


Label:  1


Кто свободен сегодня / завтра?
Есть вариант подзаработать 
Плачу 7 к


Label:  1


 ⚠️ PLATINYM Shop ⚠️

 🔴 Шишки  🔴 Амф 🔴 Меф 🔴 
Гашиш 🔴 Бошки  🔴 Экстази 🔴 Альфа 🔴 

 ⏩ Это все ты можешь найти в нашем магазине

 ✅ Также мы предоставляем:
 ⏺ Надёжный сервис
 ⏺ Отборный товар
 ⏺ Свежие и ровные адреса
 ⏺ Отзывчивый оператор
 ⏺ Магазин  ⬇️
 
 🌼 @platinynshop_bot 🌼


Label:  1


 ♾  ВАКАНСИИ В ЧАСТНУЮ КОМПАНИЮ! 

 🔻 Гибкий график: 6-8 часов в смену
 🔻 Оплата от 3000 за смену + 10% от каждого заказа
 🔻 Ежедневные выплаты 

 📩  Telegram: @prokLIZA


Label:  1


Кто свободен сегодня / завтра?
Есть вариант подзаработать 
Плачу 7 к


Label:  1


Ассалому алекум йуртдошлар
Бизда чисто 3 ойлик регистиратция 
90 кун базада туради чикиб кетмайди агар чикмаса пулларингиз 100 % кайтарилади ва 1 та  регистиратция килиш имкони бор .

           
      ☎️ +79342221717 ✅

💯✅Регистрация оргинал%✔️ 
💯✅Регистрация Квартира%
💯✅Регистрация гостиница%
💯✅Студенческий билет 💳%
💯✅Регистрация тоза  %✔️
💯✅Регистрация киргиз %🇰🇬
💯✅регистрация сохта %✔️
💯✅патент леви %✔️
💯✅патент чек леви %✔️
💯✅Перевод паспорт ор %✔️
💯✅Перевод права ор %✔️
💯✅Мед книжка тозаси ор %
💯✅Снилс оригинал %✔️
💯✅Инн оригинал %✔️
💯✅Перевод - Паспорта  ✔️
💯✅Перевод - Права                
💯✅Меграционная карта  %
💯✅Вакцина коди оргинал %
💯✅совид тест аеропорт✈️
💯✅Дипломларлар📕 %✔️
💯✅Крг ади оргинал🇰🇬 %✔️
💯✅Балничный лист %✔️
💯✅Узб права 🇺🇿🚘 %
💯✅кара права 🚍 %
💯✅рус права 🇷🇺🚘 %
💯✅Крг права🇰🇬🚘 %
💯✅Тожик права🇹🇯🚘 %
💯✅Трактир права🚜 %
💯✅доставка  метрогача
         БЕСПЛАТНО 🚘📨📮
 ☎️   +79342221717 Ходжи 
  📞  +79342221717 Ходжи 

☎️WhatsApp  +79342221717
ПРОВЕРКА РЕГИСТРАЦИЯ🔎 
ПРОВ

Label:  0


КТО СВОБОДЕН СЕЙЧАС ПЛАЧУ 3К


Label:  1


Помогy закpыть дoлги 
Быстpo и легкo 
Вcя информация у мeня


Label:  1


Помогy закpыть дoлги 
Быстpo и легкo 
Вcя информация у мeня


Label:  1


треебyются kypьеpы, самокаm


Label:  1


🏛Ассаламу Алайкум🏛
           🇹🇯 🇷🇺🇰🇬 🇰🇿 🇺🇿🇹🇲
 🕸ДЛЯ 🛗  ГРАЖДАН 🪪СНГ🌏

 ✅ 100%  
 🔠 🔠 🔠 🔠  🤦‍♂ 🔠 🔠 🔠 🔠 🔠 🔠 🔡📋

         ✅🧾РЕГИСТРАЦИЯ🧾📝 ✅
 
 ✅🏢 Квартира и 🏨 Гостиница 
🏬И Через 🤵‍♂️Фмс 👮‍♂️ Болади.  ✅
  
 ❎Без Проблем 📌Гарантия бор ✔️
🖥База 📟 Данных 💻  💯 будет  ✔️

 😀  🕓 3 ойлик Регистрация 6500 ₽ рубль  💲
 😎  📈 Первичный 🧾  2️⃣ кунга Битади 😐

🧾Регистрация ПРОДЛЕНИЯ 7000 ₽ 🏹

 🪙 ПРОДЛЕНИЯ  3️⃣ и  4️⃣ кунга Битади 😐
 🌐Подтверждения👩‍💻📰 берамиз ⬅️

📑Барча Документ📑 турлари бор
 ✅
📞 Звоните или пишите на ВАТСАП
 ☎️ 📞  +79990304040 

    🔠 🔠 🔠   🔠 🔠 🔠 🔠 🔠 🔠 🔠 🔠
 
 🔠 🔠 🔠 🔠  🔠 🔠 🔠 🔠 🔠 🔠 🔠 🔠
    
             📞 😎 😎

          https://wa.me/+79990304040


Label:  0


Ассалому алекум йуртдошлар
Бизда чисто 3 ойлик регистиратция 
90 кун базада туради чикиб кетмайди агар чикмаса пулларингиз 100 % кайтарилади ва 1 та  регистиратция килиш имкони бор .

           
      ☎️ +79342221717 ✅

💯✅Регистрация оргинал%✔️ 
💯✅Регистрация Квартира%
💯✅Регистрация гостиница%
💯✅Студенческий билет 💳%
💯✅Регистрация тоза  %✔️
💯✅Регистрация киргиз %🇰🇬
💯✅регистрация сохта %✔️
💯✅патент леви %✔️
💯✅патент чек леви %✔️
💯✅Перевод паспорт ор %✔️
💯✅Перевод права ор %✔️
💯✅Мед книжка тозаси ор %
💯✅Снилс оригинал %✔️
💯✅Инн оригинал %✔️
💯✅Перевод - Паспорта  ✔️
💯✅Перевод - Права                
💯✅Меграционная карта  %
💯✅Вакцина коди оргинал %
💯✅совид тест аеропорт✈️
💯✅Дипломларлар📕 %✔️
💯✅Крг ади оргинал🇰🇬 %✔️
💯✅Балничный лист %✔️
💯✅Узб права 🇺🇿🚘 %
💯✅кара права 🚍 %
💯✅рус права 🇷🇺🚘 %
💯✅Крг права🇰🇬🚘 %
💯✅Тожик права🇹🇯🚘 %
💯✅Трактир права🚜 %
💯✅доставка  метрогача
         БЕСПЛАТНО 🚘📨📮
 ☎️   +79342221717 Ходжи 
  📞  +79342221717 Ходжи 

☎️WhatsApp  +79342221717
ПРОВЕРКА РЕГИСТРАЦИЯ🔎 
ПРОВ

Label:  0


Ассалому алекум йуртдошлар
Бизда чисто 3 ойлик регистиратция 
90 кун базада туради чикиб кетмайди агар чикмаса пулларингиз 100 % кайтарилади ва 1 та  регистиратция килиш имкони бор .

           
      ☎️ +79342221717 ✅

💯✅Регистрация оргинал%✔️ 
💯✅Регистрация Квартира%
💯✅Регистрация гостиница%
💯✅Студенческий билет 💳%
💯✅Регистрация тоза  %✔️
💯✅Регистрация киргиз %🇰🇬
💯✅регистрация сохта %✔️
💯✅патент леви %✔️
💯✅патент чек леви %✔️
💯✅Перевод паспорт ор %✔️
💯✅Перевод права ор %✔️
💯✅Мед книжка тозаси ор %
💯✅Снилс оригинал %✔️
💯✅Инн оригинал %✔️
💯✅Перевод - Паспорта  ✔️
💯✅Перевод - Права                
💯✅Меграционная карта  %
💯✅Вакцина коди оргинал %
💯✅совид тест аеропорт✈️
💯✅Дипломларлар📕 %✔️
💯✅Крг ади оргинал🇰🇬 %✔️
💯✅Балничный лист %✔️
💯✅Узб права 🇺🇿🚘 %
💯✅кара права 🚍 %
💯✅рус права 🇷🇺🚘 %
💯✅Крг права🇰🇬🚘 %
💯✅Тожик права🇹🇯🚘 %
💯✅Трактир права🚜 %
💯✅доставка  метрогача
         БЕСПЛАТНО 🚘📨📮
 ☎️   +79342221717 Ходжи 
  📞  +79342221717 Ходжи 

☎️WhatsApp  +79342221717
ПРОВЕРКА РЕГИСТРАЦИЯ🔎 
ПРОВ

Label:  0


Кто свободен сегодня / завтра?
Есть вариант подзаработать 
Плачу 7 к


Label:  1


Ассалому алекум йуртдошлар
Бизда чисто 3 ойлик регистиратция 
90 кун базада туради чикиб кетмайди агар чикмаса пулларингиз 100 % кайтарилади ва 1 та  регистиратция килиш имкони бор .

           
      ☎️ +79342221717 ✅

💯✅Регистрация оргинал%✔️ 
💯✅Регистрация Квартира%
💯✅Регистрация гостиница%
💯✅Студенческий билет 💳%
💯✅Регистрация тоза  %✔️
💯✅Регистрация киргиз %🇰🇬
💯✅регистрация сохта %✔️
💯✅патент леви %✔️
💯✅патент чек леви %✔️
💯✅Перевод паспорт ор %✔️
💯✅Перевод права ор %✔️
💯✅Мед книжка тозаси ор %
💯✅Снилс оригинал %✔️
💯✅Инн оригинал %✔️
💯✅Перевод - Паспорта  ✔️
💯✅Перевод - Права                
💯✅Меграционная карта  %
💯✅Вакцина коди оргинал %
💯✅совид тест аеропорт✈️
💯✅Дипломларлар📕 %✔️
💯✅Крг ади оргинал🇰🇬 %✔️
💯✅Балничный лист %✔️
💯✅Узб права 🇺🇿🚘 %
💯✅кара права 🚍 %
💯✅рус права 🇷🇺🚘 %
💯✅Крг права🇰🇬🚘 %
💯✅Тожик права🇹🇯🚘 %
💯✅Трактир права🚜 %
💯✅доставка  метрогача
         БЕСПЛАТНО 🚘📨📮
 ☎️   +79342221717 Ходжи 
  📞  +79342221717 Ходжи 

☎️WhatsApp  +79342221717
ПРОВЕРКА РЕГИСТРАЦИЯ🔎 
ПРОВ

Label:  0


Pасcкажy как мoжнo зapaбoтaть денег
Бeз опытa и вложений 
Пиши мнe


Label:  1


Зaкрoй cвои кpeдиты 
Быcтро и бeз нaпpяга 
Пиши раcскaжy


Label:  1


хpyпkость


Label:  1


пpuґлyшениe


Label:  1


 💵 Два инновационных сайта предлагают уникальные криптовалютные решения. 

 ✅ Первый -  обеспечивает защиту кошельков от грязной криптовалюты, предотвращая блокировки активов.

 ✅ Второй  - генератор NFT, даря приветственные бонусы в виде пробных NFT-паков стоимостью около 30 долларов при привязке кошелька. 

 ⭐️ Перейдите на наш сайт, чтобы опробовать инновации в мире криптовалют.

 📎 AML проверка жми 📎 
 📎 Создай свой NFT жми 📎


Label:  1


🚗🚗    МОСКВАДАН ДАН    🚖🚖

🚕🚕   САНКТ-ПЕТЕРБУРГА  🚗🚗

          🚗  йолга чикяпмиз  🚕

    🚗2 та жойимиз бош одам 🚗

 📦📦  почта болса оламиз  📦📦
  
          М У Р О Ж А Т  Т ЕЛ
     
       +79690791212


Label:  0


 💵 Два инновационных сайта предлагают уникальные криптовалютные решения. 

 ✅ Первый -  обеспечивает защиту кошельков от грязной криптовалюты, предотвращая блокировки активов.

 ✅ Второй  - генератор NFT, даря приветственные бонусы в виде пробных NFT-паков стоимостью около 30 долларов при привязке кошелька. 

 ⭐️ Перейдите на наш сайт, чтобы опробовать инновации в мире криптовалют.

 📎 AML проверка жми 📎 
 📎 Создай свой NFT жми 📎


Label:  1


Нyжeн нa зaвтpa мyж нa чac oдинoкoй дaмe , выпoлнить вce oбязaннocти мyжa, зaбить гвoздь, пepeдвинyть тyмбy, пoмoчь cхoдить зa пpoдyктами , зaплaчy 3500 pyблeй, пpoшy cpoчнo oткликнyтcя


Label:  1


 💵 Два инновационных сайта предлагают уникальные криптовалютные решения. 

 ✅ Первый -  обеспечивает защиту кошельков от грязной криптовалюты, предотвращая блокировки активов.

 ✅ Второй  - генератор NFT, даря приветственные бонусы в виде пробных NFT-паков стоимостью около 30 долларов при привязке кошелька. 

 ⭐️ Перейдите на наш сайт, чтобы опробовать инновации в мире криптовалют.

 📎 AML проверка жми 📎 
 📎 Создай свой NFT жми 📎


Label:  1


 со скидкой -30% на любой товар в М.Видео 💥 
Цена 500 рублей! Без предоплат!💸 
Писать сюда —  https://t.me/mvpromik 
На этом аккаунте не отвечаю .


Label:  1


Есть  работа для вас
Работа в хорошей сфере  
Зарплата от 200$ за день
Пишите всё обсудим


Label:  1


 Тpeбyются пoмoщники на cклaд! 
Пoгpузкa и paзгрyзкa тoваpoв.
 Никaких сложнocтeй! 
Помoщь с жильeм и пpоездoм, оплaтa cpaзу пoслe смeны.
 Вce вoзpaстa пpивeтствуютcя!


Label:  


In [17]:
text_labels = zip(random_texts, labels)

In [18]:
unique_texts = {}
for text, label in text_labels:
    if text in list(unique_texts.keys()):
        continue
    else:
        unique_texts[text] = label

In [22]:
with open('mvp_texts.csv', 'w', newline='') as csvfile:
    fieldnames = ['text', 'label']
    
    writer = csv.DictWriter(csvfile, fieldnames=fieldnames, delimiter=';')

    writer.writeheader()

    for text, label in unique_texts.items():
        writer.writerow({'text': text, 'label': label})

## Without RAG

In [38]:
texts = []
labels = []
preds = []

In [39]:
with open('mvp_texts.csv', newline='') as csvfile:
    reader = csv.reader(csvfile, delimiter=';')
    for row in reader:
        texts.append(row[0])
        labels.append(row[1])

In [40]:
del(texts[0])
del(labels[0])

In [43]:
assert len(texts) == len(labels)

In [116]:
preds = []

In [117]:
chat = GigaChat(credentials='OGIyMjAxOWQtZjYwNi00NzUyLWJlMWYtYmIyMTI0ZDE4MWI0OjUwYjdkYzNmLWU5NDgtNGJjMi04ZWM5LTRkZWQ0MWNlZWI0Ng==', verify_ssl_certs=False)

messages = [
    SystemMessage(
        content='Ты профессиональный разметчик данных. Тебе нужно классифицировать каждый текст который я отправлю на 2 класса: "спам" и "не спам". Пиши мне просто "1" если текст относится к классу "спам" и "0" если текст относится к классу "не спам".'
    )
]

prompt = "Напиши 1, если этот текст относится к классу 'спам' или 0, если этот текст относится к классу 'не спам': "

for text in texts:
    messages.append(HumanMessage(content=prompt+text))
    res = chat(messages)
    messages.append(res)
    preds.append(res.content)
    time.sleep(2)

Giga generation stopped with reason: blacklist
Giga generation stopped with reason: blacklist


In [118]:
true = 0
for label, pred in zip(labels, preds):
    if label == pred:
        true += 1
print(f"Right classificated - {true}/{len(labels)}({true/len(labels)})")

Right classificated - 14/51(0.27450980392156865)


In [123]:
for text, label, pred in zip(texts, labels, preds):
    t = text.strip().replace("\n", "")
    print(f'{t}\nLabel-{label}; Pred-{pred}\n----')

ТРЕБУЕТСЯ ХАУСМЕН/ПОДСОБНЫЙ РАБОЧИЙ В ГОСТИНИЦУСтажировка оплачивается! График работы: 6/1 с 9-18 часов!Оплата: 222р в часЗаработная плата в месяц: 65/70 тысяч Питание и униформа выдаётся!Оплата 2 раза в месяц без задержки!ДЛЯ ЗАПИСИ НА ПЕРВЫЙ ДЕНЬ ПИШИТЕ В ВОТСАП ИЛИ ЗВОНИТЕ +7 905 281-37-59 Азад
Label-1; Pred-0
----
шалoвливo yлыбниcь
Label-0; Pred-0
----
Кто нуждaeтся в pабoте, в дeньгaх или закpытия дoлгoв, пиcать пoмогаю всeм
Label-1; Pred-1
----
Кто свободен сегодня / завтра?Есть вариант подзаработать Плачу 7 к
Label-1; Pred-0
----
Мальчики нyжна Вaша пoмощь Pабота не труднaя , нa пapy чаcoвПишите зa пoдробнoстями Oчeнь выpyчитe )
Label-1; Pred-0
----
‼️ СРОЧНО НУЖНЫ‼️ 💴 КУРЬЕРЫ - на доставку товара по городу, оплата от  30к в день помогаем с деньгами на расходы до первой ЗП  ✅Для деталей пишите в ЛС
Label-1; Pred-0
----
Привет 👋 ❗️Есть арбитражная cвязка  🔶✖️Без скам бирж и обменников✖️👨‍💻Набираю людей в команду работать под %Профит 2-3% с круга💸 Деньги под свое управление не бе

## With RAG

In [63]:
labels_names = {
    '0': 'не спам',
    '1': 'спам'
}

In [124]:
texts_for_rag = random.sample(list(zip(texts, labels)), 5)

In [125]:
preprocessed_texts_for_rag = ["Это пример класса '" + labels_names[label] + "' - " + text for text, label in texts_for_rag]

In [126]:
preprocessed_texts_for_rag

["Это пример класса 'спам' - исkyшeниe",
 "Это пример класса 'не спам' - 🌟Ассаламу алайкум. Хужжат таййорлаймиз.Тоза регистрация ва бошка дакументлар Арзон нархларда \n                      🇺🇿 🇹🇯 🇰🇬🇰🇿 \n\xa0 Проверка запрет депорт\n\xa0\xa0\xa0\xa0 Проверка документ\n\xa0 📘 Мед книжка оригинал \n\xa0\xa0\xa0\xa0\xa0\xa0\xa0 💳 Патент оригинал \n\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0 🧾 Чек оригинал \n\xa0 👨\u200d💻    Тоза  Регистрация \n\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0 🔖 СНИЛС\n\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0 📜 ИНН \n\xa0 💳 Права 🇺🇿🇰🇬🇷🇺🇵🇼🇹🇯\n\xa0\xa0\xa0 🎟 Айди карта 🇰🇬🇺🇿\n🎫 Кара тракторист правалар\n📇 Паспорт ,Метирка ,Права\n\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0 переводлари\n\xa0\xa0\xa0\xa0\xa0\xa0 📃 Мед справка 086\n\xa0 🇦🇲🇰🇿 Кирди Чикти печать \n\xa0\xa0\xa0\xa0\xa0 Патентга 100% отади\n\xa0\xa0 🗃 Хамма дакумент \n\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0 Ватсап  +79990304040\n\xa0\xa0\xa0\xa0\xa0\xa0\x

In [127]:
from langchain.vectorstores import Chroma
from langchain_community.embeddings import GigaChatEmbeddings

embeddings = GigaChatEmbeddings(
    one_by_one_mode=True, _debug_delay=0.005, credentials='OGIyMjAxOWQtZjYwNi00NzUyLWJlMWYtYmIyMTI0ZDE4MWI0OjUwYjdkYzNmLWU5NDgtNGJjMi04ZWM5LTRkZWQ0MWNlZWI0Ng==', verify_ssl_certs=False
)

vectordb = Chroma.from_texts(preprocessed_texts_for_rag, embeddings)

No embedding_function provided, using default embedding function: DefaultEmbeddingFunction https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2


In [128]:
docs = vectordb.similarity_search("пример класса 'не спам'", k=1)

In [129]:
docs

[Document(page_content="Это пример класса 'спам' - исkyшeниe")]

In [130]:
from langchain.chains import RetrievalQA

qa_chain = RetrievalQA.from_chain_type(chat, retriever=vectordb.as_retriever())

In [131]:
prompt = "Напиши 1, если этот текст относится к классу 'спам' или 0, если этот текст относится к классу 'не спам': "

In [132]:
rag_preds = []

In [133]:
for text in texts:
    query = prompt + text
    result = qa_chain({"query": query})['result']
    #print(result)
    rag_preds.append(result)
    time.sleep(3)

Giga generation stopped with reason: blacklist


In [134]:
true = 0
for label, pred in zip(labels, rag_preds):
    if label == pred:
        true += 1
print(f"Right classificated - {true}/{len(labels)}({true/len(labels)})")

Right classificated - 30/51(0.5882352941176471)


In [135]:
for text, label, pred in zip(texts, labels, rag_preds):
    t = text.strip().replace("\n", "")
    print(f'{t}\nLabel-{label}; Pred-{pred}\n----')

ТРЕБУЕТСЯ ХАУСМЕН/ПОДСОБНЫЙ РАБОЧИЙ В ГОСТИНИЦУСтажировка оплачивается! График работы: 6/1 с 9-18 часов!Оплата: 222р в часЗаработная плата в месяц: 65/70 тысяч Питание и униформа выдаётся!Оплата 2 раза в месяц без задержки!ДЛЯ ЗАПИСИ НА ПЕРВЫЙ ДЕНЬ ПИШИТЕ В ВОТСАП ИЛИ ЗВОНИТЕ +7 905 281-37-59 Азад
Label-1; Pred-0
----
шалoвливo yлыбниcь
Label-0; Pred-0
----
Кто нуждaeтся в pабoте, в дeньгaх или закpытия дoлгoв, пиcать пoмогаю всeм
Label-1; Pred-1
----
Кто свободен сегодня / завтра?Есть вариант подзаработать Плачу 7 к
Label-1; Pred-1
----
Мальчики нyжна Вaша пoмощь Pабота не труднaя , нa пapy чаcoвПишите зa пoдробнoстями Oчeнь выpyчитe )
Label-1; Pred-1
----
‼️ СРОЧНО НУЖНЫ‼️ 💴 КУРЬЕРЫ - на доставку товара по городу, оплата от  30к в день помогаем с деньгами на расходы до первой ЗП  ✅Для деталей пишите в ЛС
Label-1; Pred-1
----
Привет 👋 ❗️Есть арбитражная cвязка  🔶✖️Без скам бирж и обменников✖️👨‍💻Набираю людей в команду работать под %Профит 2-3% с круга💸 Деньги под свое управление не бе

# Results

Classification without RAG:

Accuracy - 27.45%

Classification with RAG:

Accuracy - 58.8%

In [4]:
import pandas as pd

In [6]:
df = pd.read_csv('twitter/train.csv')

In [51]:
ones_df = df[df['label'] == 1]

In [52]:
zeros_df = df[df['label'] == 0]

In [53]:
zeros_df = zeros_df.sample(int(zeros_df.shape[0] / 15))
ones_df = ones_df.sample(int(ones_df.shape[0] / 15))

In [54]:
zeros_df

id  label                                              tweet
15765  15766      0  #nomoneynohoney #love   #summer2016 @ psarou b...
8147    8148      0  looking forward to shopping today with karina ...
8827    8828      0    #nationalbestfriendday to these #crazy melon...
3712    3713      0   @user #worldoceansday is less than 48 hrs awa...
12904  12905      0  if you suppo trump, then you too are a racist....
...      ...    ...                                                ...
27926  27927      0  rain, cereal, coffee, naps &amp; daniel are wh...
12849  12850      0  black or clear?  black -  clear -   #earrings ...
19715  19716      0  #beaumont   gorilla simulator: you need to do ...
2720    2721      0  #sexysaturday upcoming viual love quote and te...
30906  30907      0   @user i came to realization that i am about t...

[1981 rows x 3 columns]

In [55]:
cutted_df = pd.concat([ones_df, zeros_df])

In [56]:
cutted_df

id  label                                              tweet
12628  12629      1  how is it that a man comes in last in a race b...
22529  22530      1  @user are you so bitter because you're a #mino...
19016  19017      1  i wish you &amp; your family a very #happynewy...
21031  21032      1  great aicle about #finland, #whiteness,  and #...
20971  20972      1  @user you areclueless buddy regarding long sta...
...      ...    ...                                                ...
27926  27927      0  rain, cereal, coffee, naps &amp; daniel are wh...
12849  12850      0  black or clear?  black -  clear -   #earrings ...
19715  19716      0  #beaumont   gorilla simulator: you need to do ...
2720    2721      0  #sexysaturday upcoming viual love quote and te...
30906  30907      0   @user i came to realization that i am about t...

[2130 rows x 3 columns]

In [57]:
tokens = 0
for tweet in zeros_df['tweet'].values:
    tokens += len(tweet)
tokens /= 3
print(tokens)

55110.333333333336


In [58]:
tokens = 0
for tweet in ones_df['tweet'].values:
    tokens += len(tweet)
tokens /= 3
print(tokens)

4639.0


In [61]:
cutted_df.to_csv('twitter-sentiment-analysis/cutted_data_for_labelling.csv', index=False)

In [62]:
texts = cutted_df['tweet'].values

In [63]:
texts

array(['how is it that a man comes in last in a race but he is declared a winner? #putinschoice #putinspuppet #retweetâ\x80¦ ',
       "@user are you so bitter because you're a #minoritypresident who lost the people by 3 million?  #maga #impeachtrump",
       "i wish you &amp; your family a very #happynewyear let's work together to remove #povey #xenophobia  #genderinequality from the world",
       ...,
       '#beaumont   gorilla simulator: you need to do to adapt to the environment. the need to tear the city. materi ',
       '#sexysaturday upcoming viual love quote and teaser!   #reading #amwriting #erotica #novella ',
       ' @user i came to realization that i am about to desensitized by gun violences in this country.   #prayfororlando #orlandoshoâ\x80¦'],
      dtype=object)

In [69]:
from langchain.schema import HumanMessage, SystemMessage
from langchain.chat_models.gigachat import GigaChat
import time

In [70]:
preds = []

In [76]:
len(texts)

2130

In [77]:
len(texts[132:])

1998

In [78]:
chat = GigaChat(credentials='ZThhNDlkMTktMGY0OC00NzU2LWE3YmYtZDVkYjc0MGQwODA0OjI2OWZlNTMwLWFjOWMtNDczMi05NTBjLWUzZjQzNDhlYWVkNA==', verify_ssl_certs=False)

messages = [
    SystemMessage(
        content='Ты профессиональный разметчик данных. Тебе нужно классифицировать каждый текст который я отправлю на 2 класса: "sexist or racist sentiment message" и "not sexist and not racist sentiment message". Пиши мне просто "1" если текст относится к классу "sexist or racist sentiment message" и "0" если текст относится к классу "not sexist and not racist sentiment message".'
    )
]

prompt = "Напиши 1, если этот текст относится к классу 'sexist or racist sentiment message' или 0, если этот текст относится к классу 'not sexist and not racist sentiment message': "

b_count = 0
for text in texts[132:]:
    messages.append(HumanMessage(content=prompt+text))
    res = chat(messages)
    label = res.content
    if len(str(label)) > 1:
        b_count += 1
    #messages.append(res)
    preds.append(label)
    time.sleep(1)

Giga generation stopped with reason: blacklist
Giga generation stopped with reason: blacklist


ResponseError: (URL('https://gigachat.devices.sberbank.ru/api/v1/chat/completions'), 402, b'{"status":402,"message":"Payment Required"}\n', Headers([('server', 'nginx'), ('date', 'Mon, 12 Feb 2024 21:22:30 GMT'), ('content-type', 'application/json; charset=utf-8'), ('content-length', '44'), ('connection', 'keep-alive'), ('access-control-allow-credentials', 'true'), ('access-control-allow-headers', 'Origin, X-Requested-With, Content-Type, Accept, Authorization'), ('access-control-allow-methods', 'GET, POST, DELETE, OPTIONS'), ('access-control-allow-origin', 'https://beta.saluteai.sberdevices.ru'), ('x-request-id', '7ec6c403-2cd6-4699-80c5-90cf6943606c'), ('x-session-id', 'e1062445-b54d-4eac-8095-c7fa75712087'), ('allow', 'GET, POST'), ('strict-transport-security', 'max-age=31536000; includeSubDomains'), ('allow', 'GET, POST'), ('strict-transport-security', 'max-age=31536000; includeSubDomains')]))

In [80]:
preds

['1',
 'Что-то в вашем вопросе меня смущает. Может, поговорим на другую тему?',
 '0',
 '0',
 '0',
 '1',
 '1',
 '0',
 '1',
 'Не люблю менять тему разговора, но вот сейчас тот самый случай.',
 '0',
 '0',
 '0',
 '1',
 '1',
 '0',
 '1',
 '0',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '0',
 '1',
 '0',
 '1',
 '1',
 '1',
 '1',
 '0',
 '1',
 '1',
 '1',
 '1',
 '1',
 '0',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '0',
 '0',
 '0',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '0',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '0',
 '1',
 '1',
 'Что-то в вашем вопросе меня смущает. Может, поговорим на другую тему?',
 '0'

In [81]:
for i in range(len(texts) - len(preds)):
    preds.append('ytbl')

In [82]:
len(texts) == len(preds)

True

In [83]:
cutted_df['pred'] = preds

In [84]:
cutted_df

id  label                                              tweet  \
12628  12629      1  how is it that a man comes in last in a race b...   
22529  22530      1  @user are you so bitter because you're a #mino...   
19016  19017      1  i wish you &amp; your family a very #happynewy...   
21031  21032      1  great aicle about #finland, #whiteness,  and #...   
20971  20972      1  @user you areclueless buddy regarding long sta...   
...      ...    ...                                                ...   
27926  27927      0  rain, cereal, coffee, naps &amp; daniel are wh...   
12849  12850      0  black or clear?  black -  clear -   #earrings ...   
19715  19716      0  #beaumont   gorilla simulator: you need to do ...   
2720    2721      0  #sexysaturday upcoming viual love quote and te...   
30906  30907      0   @user i came to realization that i am about t...   

                                                    pred  
12628                                                  1  
22529  Что-то в вашем вопросе меня смущает. Может, по...  
19016                                                  0  
21031                                                  0  
20971                                                  0  
...                                                  ...  
27926                                               ytbl  
12849                                               ytbl  
19715                                               ytbl  
2720                                                ytbl  
30906                                               ytbl  

[2130 rows x 4 columns]

In [89]:
list(zip(cutted_df['label'], cutted_df['pred']))[:286]

[(1, '1'),
 (1, 'Что-то в вашем вопросе меня смущает. Может, поговорим на другую тему?'),
 (1, '0'),
 (1, '0'),
 (1, '0'),
 (1, '1'),
 (1, '1'),
 (1, '0'),
 (1, '1'),
 (1, 'Не люблю менять тему разговора, но вот сейчас тот самый случай.'),
 (1, '0'),
 (1, '0'),
 (1, '0'),
 (1, '1'),
 (1, '1'),
 (1, '0'),
 (1, '1'),
 (1, '0'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '0'),
 (1, '1'),
 (1, '0'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '0'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '0'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '0'),
 (1, '0'),
 (1, '0'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 (1, '1'),
 

In [92]:
true = 0
for label, pred in zip(cutted_df['label'], cutted_df['pred']):
    if str(label) == pred:
        true += 1

In [94]:
true / 286

0.6713286713286714

In [9]:
df

id  label                                              tweet
0          1      0   @user when a father is dysfunctional and is s...
1          2      0  @user @user thanks for #lyft credit i can't us...
2          3      0                                bihday your majesty
3          4      0  #model   i love u take with u all the time in ...
4          5      0             factsguide: society now    #motivation
...      ...    ...                                                ...
31957  31958      0  ate @user isz that youuu?😍😍😍😍😍ð...
31958  31959      0    to see nina turner on the airwaves trying to...
31959  31960      0  listening to sad songs on a monday morning otw...
31960  31961      1  @user #sikh #temple vandalised in in #calgary,...
31961  31962      0                   thank you @user for you follow  

[31962 rows x 3 columns]

In [10]:
df = df[['tweet', 'label']]

In [14]:
df = df.rename({'tweet': 'text'}, axis=1)

In [17]:
df[['text']].iloc[:100].to_csv('pre_defence/data/random_texts.csv', index=False)

In [19]:
df = pd.read_csv('pre_defence/data/random_texts.csv')

In [20]:
df

text
0    @user when a father is dysfunctional and is s...
1   @user @user thanks for #lyft credit i can't us...
2                                 bihday your majesty
3   #model   i love u take with u all the time in ...
4              factsguide: society now    #motivation
..                                                ...
95  @user i'll always hope that one day i'll get t...
96  #model   i love u take with u all the time in ...
97       couple having sex fat naked japanese girls  
98  #hump on that    #hump day #humpers😩 @ edwa...
99  personalised we... gbp 7.99 get here:  #shop #...

[100 rows x 1 columns]

In [27]:
cfg = {"classes": {
        "sexist or racist sentiment message": 1,
        "not sexist and not racist sentiment message": 0
    },
"texts_path": 'pre_defence/data/random_texts.csv',
'target_column': 'text',
'save_preds_path': 'pre_defence/labelled_data/preds_no_rag.txt',
'part_start': None,
      "with_rag": False,}

In [28]:
cfg

{'classes': {'sexist or racist sentiment message': 1,
  'not sexist and not racist sentiment message': 0},
 'texts_path': 'pre_defence/data/random_texts.csv',
 'target_column': 'text',
 'save_preds_path': 'pre_defence/labelled_data/preds_no_rag.txt',
 'part_start': None,
 'with_rag': False}

In [29]:
import json 
   
   
# Convert and write JSON object to file
with open("pre_defence/configs/config_no_rag.json", "w") as outfile: 
    json.dump(cfg, outfile)

In [30]:
cfg = {"classes": {
        "sexist or racist sentiment message": 1,
        "not sexist and not racist sentiment message": 0
    },
"texts_path": 'pre_defence/data/random_texts.csv',
'target_column': 'text',
'save_preds_path': 'pre_defence/labelled_data/preds_no_rag.txt',
'part_start': None,
"with_rag": True,
"rag_texts_path": "twitter/subsample_1_percent.csv",
"k": 4}

In [32]:
cfg

{'classes': {'sexist or racist sentiment message': 1,
  'not sexist and not racist sentiment message': 0},
 'texts_path': 'pre_defence/data/random_texts.csv',
 'target_column': 'text',
 'save_preds_path': 'pre_defence/labelled_data/preds_no_rag.txt',
 'part_start': None,
 'with_rag': True,
 'rag_texts_path': 'twitter/subsample_1_percent.csv',
 'k': 4}

In [33]:
import json 
   
   
# Convert and write JSON object to file
with open("pre_defence/configs/config_rag.json", "w") as outfile: 
    json.dump(cfg, outfile)

In [ ]:
classes = {
        'relevant to the United States Economy': 1,
        'not relevant to the United States Economy': 0
    }

#texts_path = 'us_news/dataset.csv'
#target_column = 'text'
#save_preds_path = 'us_news/rag_1/preds.txt'
#part_start = None

with_rag = True
rag_texts_path = 'us_news/subsample_1_percent.csv'
k = 4